src/utils/self_test/self_test.c

```diff
@@ -434,72 +695,1112 @@ static void print_results(struct st_latency *latencies,
+						 percent_diff);
+					if (value >= upper) {
+						passed = false;
+						results = "Failed:";
+						/* ret_value = -1; */
+					}
+				} else if (status[i].flag & TST_LOW) {
+					snprintf(range, RANGE_SIZE,
+						 " Range (%6d --   "
+						 "  ..)   %3d%%"
+						 " %5.1f%%",
+						 (int)lower, (int)scale,
+						 percent_diff);
+					if (value <= lower) {
+						passed = false;
+						results = "Failed:";
+						/* ret_value = -1; */
+					}
+				}
+
+				/* print results */
+				if (firstpass)
+					printf("\n Endpoint Result (%s)\n",
+					       status[i].description);
+				firstpass = false;
+				printf("   %s : %8d  %s %s\n",
+				       key, (int)value, results, range);
+				snprintf(compar_result, sizeof(compar_result),
+					 "%s %8d %s",
+					 results, (int)value, range);
+
+				/* Log error message */
+				if (!passed) {
+					/* avoid checkpatch warning */
+					D_INFO("%s %s range check\n",
+					       key, results);
+				}
+
+				/* Place results into result section */
+				ConfigAddString(cfg_output, result_section_name,
+						key, compar_result);
+increment_code:
+				/* loop control:*/
+				j++;
+				if (j == number_keys) {
+					j = 0;
+					number_keys = ConfigGetKeys(cfg_output,
+							    section_name,
+							    key_names, max,
+							    total_number_keys);
+					total_number_keys += number_keys;
+				}
+			}
+		} /* end of valid test loop */
+	} /* end of "for" loop */
+cleanup:
+	if (cfg_expected != NULL)
+		ConfigFree(cfg_expected);
+	return ret_value;
+}
+
+/* Place output results into output file, tagging with master key */
+static int combine_results(Config *cfg_results, char *section_name)
+{
+	ConfigRet	 ret;
+	int		 ret_value = 0;
+	int		 temp;
+	int		 temp2;
+	int		 i;
+	int		 dfault = -1;
+	const char	*key_name;
+	size_t		 size = MASTER_VALUE_SIZE;
+	char		 new_key_name[2 * MASTER_VALUE_SIZE];
+	char		 master[MASTER_VALUE_SIZE];
+	int		 status_size = sizeof(status) /
+				       sizeof(struct status_feature);
+
+	/* Read master-endpoint string */
+	ret = ConfigReadString(cfg_results, section_name, "master_endpoint",
+			       master, size, "ME");
+	/*
+	 * Copy results into output.
+	 * If the output already has it, then remove it.
+	 */
+	for (i = 0; i < status_size; i++) {
+		key_name = status[i].name;
+		ret = ConfigReadInt(cfg_results, section_name, key_name,
+				    &temp, dfault);
+		/* Tag master endpoint to key name */
+		snprintf(new_key_name, sizeof(new_key_name),
+			 "%s-%s", master, key_name);
+
+		/* Check to see if key occurres in results */
+		if (ret == CONFIG_OK) {
+			/* Check to see if key already exist in output */
+			ret = ConfigReadInt(cfg_output, section_name,
+					    key_name, &temp2, dfault);
+			if (ret == CONFIG_OK) {
+				ConfigRemoveKey(cfg_output, section_name,
+						key_name);
+			}
+			ConfigAddInt(cfg_output, section_name,
+				     new_key_name, temp);
+		}
+	}
+
+#ifdef PRINT_IMMEDIATE_RESULTS
+#define PRINT_IMMEDIATE_RESULTS
```

Comment:
this is strange code. why define if it is already defined?

Reply:
The #define is within the #ifdef so it is undefined.  
 It's just a simple way I have to enable defines when I need it for debugging.  It is much simplier to move the define statement above the if statement then it is to retype the entire #define statement.
I can remove this is you like.